includes/wp-cli/class-orchestrate-runner.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+namespace Automattic\WP\Cron_Control\CLI;
+
+/**
+ * Commands used by the Go-based runner to execute events
+ */
+class Orchestrate_Runner extends \WP_CLI_Command {
+	/**
+	 * List the next set of events to run; meant for Runner
+	 *
+	 * Will not be all events, just those atop the curated queue
+	 *
+	 * Not intended for human use, rather it powers the Go-based Runner. Use the `events list` command instead.
+	 *
+	 * @subcommand list-due-batch
+	 */
+	public function list_due_now( $args, $assoc_args ) {
+		if ( 0 !== \Automattic\WP\Cron_Control\Events::instance()->run_disabled() ) {
+			\WP_CLI::error( __( 'Automatic event execution is disabled', 'automattic-cron-control' ) );
+		}
+
+		$events = \Automattic\WP\Cron_Control\Events::instance()->get_events();
+
+		$format = \WP_CLI\Utils\get_flag_value( $assoc_args, 'format', 'table' );
+
+		\WP_CLI\Utils\format_items( $format, $events['events'], array(
+			'timestamp',
+			'action',
+			'instance',
+		) );
+	}
+
+	/**
+	 * Run a given event; meant for Runner
+	 *
+	 * Not intended for human use, rather it powers the Go-based Runner. Use the `events run` command instead.
+	 *
+	 * @subcommand run
+	 * @synopsis --timestamp=<timestamp> --action=<action-hashed> --instance=<instance>
+	 */
+	public function run_event( $args, $assoc_args ) {
+		if ( 0 !== \Automattic\WP\Cron_Control\Events::instance()->run_disabled() ) {
+			\WP_CLI::error( __( 'Automatic event execution is disabled', 'automattic-cron-control' ) );
+		}
+
+		$timestamp = \WP_CLI\Utils\get_flag_value( $assoc_args, 'timestamp', null );
+		$action    = \WP_CLI\Utils\get_flag_value( $assoc_args, 'action',    null );
+		$instance  = \WP_CLI\Utils\get_flag_value( $assoc_args, 'instance',  null );
+
+		if ( ! is_numeric( $timestamp ) || ! is_string( $action ) || ! is_string( $instance ) ) {
```

Comment:
Good call. Goes along with #124 and related comments here.

> makes it far easier to fix issues when you know exactly what's wrong

But that takes all the fun out of debugging! ;)